### DISCUSSION TITLE
Is there a way to do a where clause to check for a column typed UUID?

### DISCUSSION THREAD

**Discussion by lukteo** [go-jet/jet#245] [Q&A]
Hello, thanks for the great work with go-jet! I have a question which is exactly that of the title. 

I am using postgres as my database and would like to get users based on their id. In my migration I am using uuid as the the type for the ID column, but during query I realize .EQ() only supports string expressions and though I was able to convert it, I was receiving errors of "jet: pq: operator does not exist: uuid = text". 

What would you suggest be the right way to do a where clause with a uuid column ?

**Comment by go-jet** [comment]
Hi @teoyi ,
Did you try `UUID`? 
`Table.ID.EQ(UUID("1a4b-...."))`

**Comment by philon123** [comment]
I'm running into the same error as @lukteo with this line:
`.WHERE(table.User.ID.EQ(pg.String(input.ID.String())))`

 @go-jet your recommendation is not working unfortunately. The folllowing fails with a compiler error, since it tries to import the internal package `github.com/go-jet/jet/v2/internal/jet`. 
`.WHERE(table.User.ID.EQ(jet.UUID(input.ID)))`

It would be great if we didn't have to deal with strings at all! Is any workaround known for now?

**Reply by lukteo** [reply]
Hey @philon123 

Just wanted to see what was your setup. Is your `User.ID` of type `uuid.UUID` ? or is it an actual `string` type ? 

Eitherway, here is what worked for me based on the recommendation maybe it will be of help to you. 

```go
import ( 
    . "github.com/go-jet/jet/v2/postgres"
)

func ... (
    targetID uuid.UUID, 
) error { 
    ... 
    WHERE(tbl.ID.EQ(UUID(targetID))
    ...
}
```

**Reply by go-jet** [reply]
@philon123 You need to import db specific package(like the postgres in @lukteo example), not internal jet package.

**Reply by philon123** [reply]
Thanks @lukteo and @go-jet ! Indeed my `User.ID` is a UUID. Your suggested code works perfectly. My mistake was to copy the example provided by @go-jet earlier: `Table.ID.EQ(UUID("1a4b-...."))` and just guessing that UUID must be in package `jet`. I could probably have looked around more for better candides to import.
